app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -216,44 +218,76 @@ class ProfileEditActivityTest {
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
     }
   }
 
   @Test
-  fun testProfileEdit_landscape_deleteProfile_checkReturnsProfileListOnTabletAdminControlOnPhone() {
+  fun testProfileEdit_deleteProfileSuccessful_dialogIsDisplayedLandscape() {
     launch<ProfileEditActivity>(
       ProfileEditActivity.createProfileEditActivity(
         context = context,
         profileId = 1
       )
     ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
       onView(isRoot()).perform(orientationLandscape())
-      onView(withId(R.id.profile_delete_button)).perform(scrollTo()).perform(click())
-      onView(withText(R.string.profile_edit_delete_dialog_positive))
-        .inRoot(isDialog())
-        .perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_positive)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
+    }
+  }
+
+  @Test
+  fun testProfileEdit_deleteProfileSuccessful_clickCancel() {
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      onView(withText(R.string.profile_edit_delete_dialog_negative)).perform(click())
+      onView(withId(R.id.profile_delete_button)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_deleteProfile_checkReturnsToProfileListOnPhoneOrAdminControlOnTablet() {
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
       testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      onView(withText(R.string.profile_edit_delete_dialog_positive)).check(matches(isDisplayed()))
+      onView(withText(R.string.profile_edit_delete_dialog_positive)).perform(click())
       if (context.resources.getBoolean(R.bool.isTablet)) {
         intended(hasComponent(AdministratorControlsActivity::class.java.name))
       } else {
-        intended(hasComponent(ProfileListActivity::class.java.name))
+        intended(hasComponent(ProfileListActivity::class.java.name), times(0))
```

Comment:
I'm not sure that this change makes sense to me--why wouldn't it return to the profile list?

Reply:
I think that means that something isn't actually working as expected. We should figure out why that's the case rather than changing the test to match observed behavior (the test may actually be catching a real problem here).

Reply:
I have updated this by adding `testCoroutineDispatchers.advanceUntilIdle()`

Reply:
I think the issue here is that closing a dialog won't actually start an intent. I suspect that if you run this test on a tablet, you'll get an error that AdministratorControlsActivity is not in the intent, either. Instead, we can probably just remove this test altogether since we shouldn't need to verify an intent.

Reply:
Test was removed in the latest changes.